File: pkg/R/SparkRBackendInterface.R
#
# Definitions of methods implemented by SparkRBackend
# Corresponding Java definition found in SparkRBackendInterface.scala
#

createSparkContext <- function(
  master,
  appName,
  sparkHome,
  jars,
  sparkEnvirMap=list(),
  sparkExecutorEnvMap=list()) {

  # Find a way to pass hash map?
  invokeJava("createSparkContext", master, appName, sparkHome,
             jars, sparkEnvirMap, sparkExecutorEnvMap)
}

stopBackend <- function() {
  invokeJava("stopBackend")
  invisible(NULL)

  # Also close the connection and remove it from our env
  close(conn)
  rm(".sparkRCon", envir=.sparkREnv)
}

File: pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SparkRBackendInterface.scala
package edu.berkeley.cs.amplab.sparkr

// Definitions of methods implemented by SparkRBackend
// Corresponding definition found in SparkRBackendInterface.R
// 
// NOTE: If you change any method here, you will need to change the
// corresponding one in SparkRBackendInterface.R
trait SparkRBackendInterface {

  /**
   * Create a Spark Context
   *
   * @return String the applicationId corresponding to this SparkContext
   *
   */
  def createSparkContext(master: String, appName: String, sparkHome: String,
    jars: Array[String], sparkEnvirMap: Map[String, String],
    sparkExecutorEnvMap: Map[String, String]): String

  // def saveAsTextFile(rdd: Int, path: String): Int

  // def saveAsObjectFile(rdd: Int, path: String): Int
}
